### PR TITLE
PP-941 Adding live provider configurations

### PR DESCRIPTION
**Pull request by rahul-vinayak**
- New column in gateway_accounts table called ‘type’ which basically
  tells whether the account type is a ‘Test’ one or a ‘Live’ one.
- New configurations in the config.yaml specifying the live or test
  provider url configurations. Which url is used depends on which account
  is used to generate the API key, which is based on the ‘type’ column in
  the gateway_accounts table above.
- Additions in migration.xml for liquibase changes
- Modified and fixed failing tests to suffice with the above two
  changes.
- Ansible and pay-script changes to follow
- TBM after scripts and ansible

### CODE REVIEW

File: src/main/resources/migrations.xml [hunk 1]
```diff
@@ -349,4 +349,13 @@
         </createIndex>
     </changeSet>
 
+    <changeSet id="add type(TEST/LIVE) column to gateway_accounts table" author="shakespeare">
```
Comment (ckalista) [inline src/main/resources/migrations.xml]: Author Shakespeare ? :-)

Comment (rahul-vinayak) [inline src/main/resources/migrations.xml]: that was my easter egg, and to check who is paying attention :)

**Comment by ckalista** [comment]
Code review completed. No further comments.

File: src/main/resources/migrations.xml [hunk 2]
```diff
@@ -349,4 +349,13 @@
...
+        <addColumn tableName="gateway_accounts">
+            <column name="type" type="varchar(10)" defaultValue="TEST" remarks="possible values (TEST/LIVE)">
+                <constraints nullable="false" unique="false"/>
+            </column>
+        </addColumn>
+        <addUniqueConstraint tableName="gateway_accounts" columnNames="payment_provider, type"/>
```
Comment (ijmad) [inline src/main/resources/migrations.xml]: This constraint is wrong - we will have many live/test worldpay accounts, many sandbox accounts, etc. This constraint will prevent that and will not apply even on the existing contents of the gateway_accounts table in integration. This must be removed.

**Comment by ijmad** [comment]
This must NOT be merged until the issue with the constraints is fixed.

File: src/main/resources/migrations.xml (hunk 1 above)
Comment (ijmad) [inline src/main/resources/migrations.xml]: I appreciate a cute easter egg as much as the next dev, but please don't do this - undocumented features are an issue in PCI-DSS, and while this is harmless we should set good principles please.

File: src/main/resources/migrations.xml (hunk 2 above)
Comment (rahul-vinayak) [inline src/main/resources/migrations.xml]: Constraint removed now

**Comment by rahul-vinayak** [comment]
As discussed, removed the Unique key constraint now and merge shouldn't be stopped for that reason anymore.

**Pull request merged**